src/include/threadpool/logger_queue_pool.h

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// mono_queue_pool.h
+//
+// Identification: test/threadpool/mono_queue_pool.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+#include "task.h"
+#include "logger.h"
+
+// TODO: tune these variables
+#define DEFAULT_LOGGER_POOL_SIZE 1
+#define DEFAULT_TASK_QUEUE_SIZE 20000
+
+namespace peloton {
+namespace threadpool {
+/**
+ * @class MonoQueuePool
+ * @brief Wrapper class for single queue and single pool
+ * One should use this if possible.
+ */
+class LoggerQueuePool {
```

Comment:
I think this is the part that leads the overall test coverage to drop. Can you reuse the code in the previously implemented thread pool infrastructure? We can change the thread pool's code to avoid singleton if this could help you.

Reply:
Yes, singleton is bad because those tasks (worker and logger) are not supposed to mix up, therefore we need to be able to have 2 instances of the same structure. Since all the functions are generic in their parameters, I also think this would work fine.

With the ability of instatiating different pools, we can define each one a different number of threads, I suppose?

Reply:
Yes. Let me check with @dengyuchenkit and see if she could do this.